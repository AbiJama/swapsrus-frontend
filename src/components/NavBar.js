import React from "react";

import { Link } from "react-router-dom";
import "../styles/navbar.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faUser } from "@fortawesome/free-solid-svg-icons";

function NavBar() {
  return (
    <div className="nav-bar">
      <img className="logo" src="logo2.png" alt="swapzRus icon" />
      <ul className="navbar-links">
        <li className="navbar-links-item">
          <Link to="/about-us">About Us</Link>
        </li>
        <li className="navbar-links-item">
          <Link to="/toys">Browse Toys</Link>
        </li>
        <li className="navbar-links-item">
          <Link to="/add-toys">Add Toys</Link>
        </li>
        <li className="navbar-profile-icon">
          <Link to="/profile">
            Account <FontAwesomeIcon icon={faUser} />
          </Link>
          <Link to="/login">
            Login
          </Link>
          <Link to="/register">
            Register
          </Link>
        </li>
      </ul>
    </div>
  );
}

export default NavBar;
