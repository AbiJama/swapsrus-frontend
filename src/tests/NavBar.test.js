import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import NavBar from "../components/NavBar";

describe("NavBar", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );

    const logo = screen.getByAltText("swapzRus icon");
    const aboutLink = screen.getByRole("link", { name: /About Us/i });
    const toysLinks = screen.getAllByRole("link", { name: /Toys/i });
    const addToysLink = screen.getByRole("link", { name: /Add Toys/i });

    expect(asFragment()).toMatchSnapshot();
    expect(logo).toBeInTheDocument();
    expect(aboutLink).toBeInTheDocument();
    expect(toysLinks).toHaveLength(2);
    expect(addToysLink).toBeInTheDocument();
  });
});
