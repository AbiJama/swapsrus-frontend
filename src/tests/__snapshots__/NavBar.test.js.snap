// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <img
      alt="swapzRus icon"
      class="logo"
      src="logo2.png"
    />
    <ul
      class="navbar-links"
    >
      <li
        class="navbar-links-item"
      >
        <a
         
          href="/about-us"
        >
          About Us
        </a>
      </li>
      <li
        class="navbar-links-item"
      >
        <a
          
          href="/toys"
        >
          Browse Toys
        </a>
      </li>
      <li
        class="navbar-links-item"
      >
        <a
          
          href="/add-toys"
        >
          Add Toys
        </a>
      </li>
      <li
        class="navbar-links-item"
      >
        <a
          class=""
          href="/login"
        >
          Login
        </a>
      </li>
      <li
        class="navbar-links-item"
      >
        <a
          class=""
          href="/register"
        >
          Register
        </a>
      </li>
    </ul>
    <ul>
      <li
        class="navbar-profile-icon"
      >
        <a
          href="/profile"
        >
          Account   
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-user "
            data-icon="user"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z"
              fill="currentColor"
            />
          </svg>
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
